src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -249,10 +249,10 @@ internal sealed class SourcePropertySymbol : PropertySymbol, IAttributeTargetSym
                             CustomModifierUtils.CopyTypeCustomModifiers(overriddenPropertyType.Type, type.Type, this.ContainingAssembly),
                             overriddenPropertyType.CustomModifiers);
 
-                        // Although we only do this in error scenarios, it is undesirable to mutate the symbol by setting its type twice.
+                        // It is undesirable to mutate the symbol by setting its type twice.
                         // Tracked by https://github.com/dotnet/roslyn/issues/35381
-                        _lazyType.InterlockedDangerousReset();
-                        _lazyType.InterlockedInitialize(type);
+                        _lazyType = null; // a regular assignment is fine because we're in the constructor (we have not handed the instance to other threads)
```

Comment:
I don't think we can fix this now, but calling instance methods implicitly leaks the object. I don't think we have any guarantee that we're not exposing this symbol to multiple threads.